### PR TITLE
Sample usage of C# default interface implementation together with tryParse operator

### PR DESCRIPTION
**Pull request by wallymathieu**

### CODE REVIEW

File: tests/FSharpPlus.Tests/General.fs [hunk 1]
```diff
@@ -2237,7 +2244,7 @@ module Sequences =
           let! z = seq [1..50]
           let! x = seq [1..z]
           let! y = seq [x..z]
-          if (x*x + y*y = z*z) then return (x, y, z)}
+          if (x*x + y*y = z*z) then return (x, y, z) else () }
```
Comment (wallymathieu) [inline tests/FSharpPlus.Tests/General.fs:2247]: This looks like a change in F# 5

File: tests/FSharpPlus.Tests/General.fs [hunk 2]
```diff
@@ -1784,6 +1784,7 @@ module MonadTransformers =
         let _ = put initialState : ChoiceT<State<int, Choice<unit,string>>>
 
         ()
+#if !NETSTANDARD3_0
```
Comment (wallymathieu) [inline tests/FSharpPlus.Tests/General.fs:1787]: These examples does not compile with F# 5, so are probably regressions

**Comment by wallymathieu** [comment]
Note the changes when compiling with F# 5. The type inference for CE builders seems to be a breaking change, unsure if it's intentional. @dsyme 
Note that these are the same changes became evident for in the PR that updated F# formatting:
https://github.com/fsprojects/FSharpPlus/pull/372

**Comment by gusty** [comment]
I created a standalone repro and submitted it https://github.com/dotnet/fsharp/issues/10379

File: tests/CSharpLib/CustomerId.cs [hunk 3]
```diff
@@ -23,4 +26,17 @@ public static bool TryParse(string value,out CustomerId id)
             return false;
         }
     }
+#if NETSTANDARD2_1 
+    public interface ICustomerId{
```
Comment (gusty) [inline tests/CSharpLib/CustomerId.cs:30]: We don't have guidelines for C# in this projects, but at least could you leave a space after `;` and surround with spaces `{`, , `}` and `=`, which I think is kind of standard in both languages?

**Review by gusty — APPROVED** [review]
Cool, you can merge it, after the suggested change.

File: tests/CSharpLib/CustomerId.cs (hunk 3 above)
Comment (wallymathieu) [reply tests/CSharpLib/CustomerId.cs]: As usual I'm a bit sloppy about indentation. Thanks!

**Pull request merged**